src/components/pagination.tsx

```diff
@@ -47,7 +48,32 @@ export const Pagination: StatelessComponent<Props> = ({
   query,
   pageLinkBuilder,
   renderHead,
+  onButtonClick,
 }) => {
+  const [currentPage, setCurrentPage] = useState(null)
+  const [navButtonClicked, setNavButtonClicked] = useState("")
+
+  const handlePageChange = (data) => {
+    setCurrentPage(data)
+  }
+
+  const handleNavButtonClick = (label) => {
+    setNavButtonClicked(label)
+  }
+
+  useEffect(() => {
+    if (currentPage) {
+      if (navButtonClicked && onButtonClick) {
+        onButtonClick(navButtonClicked)
+      } else if (currentPage && onButtonClick) {
+        onButtonClick(`${currentPage.selected + 1}`)
+      }
+      onPageChange(currentPage)
+      setNavButtonClicked("")
+    }
+  }, [currentPage])
```

Comment:
you are missing some dependencies of the hook here

Reply:
Already discussed with @talamcol in slack about this one. We decided that its best to keep the deps array as is in this use case.